### PR TITLE
workaround issue with the `scrollBetweenSections`

### PR DESCRIPTION
**Pull request by vabarbosa**
Related: https://github.com/Qiskit/qiskit.org/issues/2411

this PR is a workaround for the issue of the `scrollBetweenSections` not working when using `nuxt generate`.
it also reverts the temporary change from https://github.com/Qiskit/qiskit.org/pull/2420

preview url: https://qiskit-org-pr-2427.dcq4xc5i083.us-south.codeengine.appdomain.cloud/

---
the main issue appears to be with the `.scrollable` component getting reinitialized/rerendered after the `InspectionObserver` has been initially set. as a result, the callback (from scrolling) is not happening since the original node no longer exists

### CODE REVIEW

**Comment by vabarbosa** [comment]
@korgan00 any thoughts on the issue here and a perhaps a better way to resolve instead of just [adding a timeout/delaying the InspectionObserver](https://github.com/Qiskit/qiskit.org/pull/2427/files#diff-eed58d8ecfd1765c3f93845d380926fff3cf3e278e9c95b9331083c083318290R20-R27)?

**Comment by korgan00** [comment]
> @korgan00 any thoughts on the issue here or perhaps a better way to resolve (instead of just [adding a timeout/delaying the InspectionObserver](https://github.com/Qiskit/qiskit.org/pull/2427/files#diff-eed58d8ecfd1765c3f93845d380926fff3cf3e278e9c95b9331083c083318290R20-R27))?

I am thinking about it. I want to be able o test it on my computer. I prefer the [`nextTick`](https://vuejs.org/v2/api/#Vue-nextTick), but I don't know if it will work here. Also, you can call the `updateObserved` method to simplify the code.

**Comment by vabarbosa** [comment]
> > @korgan00 any thoughts on the issue here or perhaps a better way to resolve (instead of just [adding a timeout/delaying the InspectionObserver](https://github.com/Qiskit/qiskit.org/pull/2427/files#diff-eed58d8ecfd1765c3f93845d380926fff3cf3e278e9c95b9331083c083318290R20-R27))?
> 
> I am thinking about it. I want to be able o test it on my computer. I prefer the [`nextTick`](https://vuejs.org/v2/api/#Vue-nextTick), but I don't know if it will work here. Also, you can call the `updateObserved` method to simplify the code.

yeah, i tried `nextTick` but it did not work.

File: mixins/scrollBetweenSections.ts [hunk 1]
```diff
@@ -16,12 +16,12 @@ export default class ScrollBetweenSections extends Vue {
...
-      .querySelectorAll('.scrollable')
-      .forEach((section) => {
-        (this._observer as IntersectionObserver).observe(section)
-      })
+    )
+    this.updateObserved()
+  }
+
+  updated () {
+    this.$nextTick(() => this.updateObserved())
```
Comment (vabarbosa) [inline mixins/scrollBetweenSections.ts:24]: @techtolentino @korgan00 i think i got it! i was able to resolve the issue using `nextTick` within the `updated` hook. feel free to review, test, check out the [preview link](https://qiskit-org-pr-2427.dcq4xc5i083.us-south.codeengine.appdomain.cloud/) and let me know if you see any problems or have any concerns. thanks.

Comment (techtolentino) [reply mixins/scrollBetweenSections.ts:24]: this looks good to me, but I'll wait for @korgan00 

👏🏽 

thanks @vabarbosa

Comment (korgan00) [reply mixins/scrollBetweenSections.ts:24]: Yeah! It looks so good to me. I always forget the `updated` callback.

Amazing! Thank you!

**Pull request merged**